### PR TITLE
[IMP] mail: Delete tracking values when uninstalling module

### PR DESCRIPTION
**Pull request by LucasLefevre**
[IMP] mail: Delete tracking values when uninstalling module
--------------
### Purpose

mail.tracking.values should be deleted when the corresponding field is deleted (when the module which defined it is uninstalled).

Firstly because we don't want useless data in db.
Secondly because the groups associated with the field can no longer be checked if it was deleted. As we don't know to whom the value was restricted, the value should not be displayed anyway.

Note: this case was fixed in saas-12.2 by b9e96b7 but the proper
way to fix it is to delete the tracking values.

### Specification

Delete `mail.tracking.value` when the associated `ir.model.fields` is deleted.

Two alternatives were considered:

1. Change the `field` field of `mail.tracking.value` from Char to a m2o to `ir.model.fields` which allows to use delete oncascade. This implies to modify existing code, but more importantly it adds database queries.

2. Override the unlink method of `ir.model.fields` to first unlink associated tracking values.

The first method is probably cleaner but the second method was nonetheless chosen as we don't want to impact performance in the main tracking flow only to better support module uninstalls which happens rarely.

Note: when a module is uninstalled, `ir.model.fields` are unlinked one by one (in a for loop). Tracking values are therefore also unlinked field by field. Batchifying field deletion would greatly reduce the amount of queries.

[IMP] mail: Rename groups field of mail.tracking.value
----------------

This commit renames the `groups` field into `field_groups`.

The previous name was impossible to grep as the word `groups` is everywhere in the code base.

Also the new name better follows the implicit convention that fields related to the tracked field all start
with `field_`.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/models/mail_tracking_value.py [hunk 1]
```diff
@@ -6,6 +6,28 @@
 from odoo import api, fields, models, tools
 
 
+class IrModelField(models.Model):
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_tracking_value.py]: Small note: there is already an ir model fields override in ir_model.py .

How to gain 22 points: split file into 2 (becayse @tivisse is too shy to do it)

How to gain 222 points: make a small test

**Comment by tivisse** [comment]
@robodoo r+ merge

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo rebase-ff

**Comment by tivisse** [comment]
@robodoo merge

**Pull request merged**